src/code/z_demo.c

```diff
@@ -1,99 +1,1650 @@
+                for (j = 0; j < cmdEntries; j++) {
+                    cmd = (CsCmdBase*)cutscenePtr;
+                    if ((csCtx->frames >= cmd->startFrame) && (csCtx->frames < cmd->endFrame)) {
+                        csCtx->playerAction = (CsCmdActorAction*)cutscenePtr;
+                    }
+                    cutscenePtr += sizeof(CsCmdActorAction);
+                }
+                break;
+
+            case CS_CMD_CAMERA:
+                cutscenePtr = &cutscenePtr[Cutscene_Command_Camera(globalCtx, cutscenePtr)];
+                break;
+
+            case CS_CMD_TERMINATOR:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_Terminator(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_15F:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    func_800EB6F8(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_TEXTBOX:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    cmd = (CsCmdBase*)cutscenePtr;
+                    if (cmd->base != 0xFFFF) {
+                        Cutscene_Command_Textbox(globalCtx, csCtx, (CsCmdTextbox*)cutscenePtr);
+                    }
+                    cutscenePtr += sizeof(CsCmdTextbox);
+                }
+                break;
+
+            case CS_CMD_SCENE_TRANS_FX:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_TransitionFX(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_MOTIONBLUR:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_MotionBlur(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case CS_CMD_GIVETATL:
+                bcopy(cutscenePtr, &cmdEntries, 4);
+                cutscenePtr += 4;
+                for (j = 0; j < cmdEntries; j++) {
+                    Cutscene_Command_GiveTatlToPlayer(globalCtx, csCtx, (CsCmdBase*)cutscenePtr);
+                    cutscenePtr += sizeof(CsCmdBase);
+                }
+                break;
+
+            case -2:
+                // Very useful and necessary case
+                break;
```

Comment:
```suggestion
            case -2:
                break;
```
This comment can probably go. Or be replace by an explanation of where -2 comes from and why it's a do-nothing case. It seems to be related to `D_801F4DC8`/actor action commands, and handles those commands separately instead of through this switch

Reply:
Oh yeah, that was just a joke I forgot to remove before PRing :sweat_smile: